consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/blockcreation/BftMiningCoordinator.java

```diff
@@ -90,13 +91,19 @@ public void stop() {
       blockchain.removeObserver(blockAddedObserverId);
       bftProcessor.stop();
       // Make sure the processor has stopped before shutting down the executors
-      try {
-        bftProcessor.awaitStop();
-      } catch (final InterruptedException e) {
-        LOG.debug("Interrupted while waiting for IbftProcessor to stop.", e);
-        Thread.currentThread().interrupt();
-      }
-      bftExecutors.stop();
+      // TODO this may not complete before the next miningCoordinator is started
+      // workaround until this is complete: https://github.com/hyperledger/besu/issues/3003
+      Executors.newSingleThreadExecutor()
```

Comment:
This is changing the behaviour from synchronous to asynchronous. It will workaround the thread issue though.
Not sure what negative impacts this will have yet. But wanted to raise this as a potential issue.

Reply:
For sure, I was just putting something in to enable local testing. I shouldn't include this in the merged PR though before we resolve #3003 so good catch.

Reply:
I'm currently thinking #3003 fix could still be async but with an await (e.g. Future/CompletableFuture).
It got a little hairy when I was playing with it, so aborted.

Reply:
I made this synchronous again